shopfloor/actions/inventory.py

```diff
@@ -59,7 +59,7 @@ def _create_draft_inventory(self, location, product, package=None, lot=None):
             for quant in quants:
                 if quant.inventory_quantity_set:
                     continue
-                quants.write(
+                quant.write(
                     {
                         # Set an inventory quantity to prevent the zero quant cleanup
                         "inventory_quantity": quant.inventory_quantity + 1,
```

Comment:
Putting `user_id` value here will allow to protect zero quants with confirmation being deleted by Odoo:

https://github.com/odoo/odoo/blob/16.0/addons/stock/models/stock_quant.py#L926

Reply:
Pushed a new commit, your review is welcome @rousseldenis